ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStoreV2.java

```diff
@@ -63,10 +78,100 @@ public CloudBlobStoreV2(VerifiableProperties properties, MetricRegistry metricRe
     this.blobContainerClient = blobContainerClient;
   }
 
+  public static final String ACCOUNT_ID = "accountId";
+  public static final String CONTAINER_ID = "containerId";
+  public static final String CRC = "crc";
+  public static final String EXPIRATION_TIME = "expirationTimeInMs";
+  public static final String LIFE_VERSION = "lifeVersion";
+  public static final String OPERATION_TIME = "operationTimeMs";
+  public static final String SIZE = "size";
+
+  /**
+   * @return a {@link HashMap} of metadata key-value pairs.
+   */
+  public Map<String, String> messageInfoToMap(MessageInfo messageInfo) {
+    HashMap<String, String> hashMap = new HashMap<>();
+    hashMap.put(ACCOUNT_ID, String.valueOf(messageInfo.getAccountId()));
+    hashMap.put(CONTAINER_ID, String.valueOf(messageInfo.getContainerId()));
+    hashMap.put(CRC, String.valueOf(messageInfo.getCrc()));
+    hashMap.put(EXPIRATION_TIME, String.valueOf(messageInfo.getExpirationTimeInMs()));
+    hashMap.put(LIFE_VERSION, String.valueOf(messageInfo.getLifeVersion()));
+    hashMap.put(OPERATION_TIME, String.valueOf(messageInfo.getOperationTimeMs()));
+    hashMap.put(SIZE, String.valueOf(messageInfo.getSize()));
+    return hashMap;
+  }
+
+  /**
+   * Uploads Ambry blob to Azure blob storage.
+   * If blob already exists, then it catches BLOB_ALREADY_EXISTS exception and proceeds to next blob.
+   * It fails for any other exception.
+   * @param messageSetToWrite The message set to write to the store
+   *                          Only the StoreKey, OperationTime, ExpirationTime, LifeVersion should be used in this method.
+   * @throws StoreException
+   */
   @Override
   public void put(MessageWriteSet messageSetToWrite) throws StoreException {
```

Comment:
Old code has sanity check.
Is that safe to remove them?

```
    if (messageSetToWrite.getMessageSetInfo().isEmpty()) {
      throw new IllegalArgumentException("Message write set cannot be empty");
    }
    checkDuplicates(messageSetToWrite.getMessageSetInfo());
```

Reply:
1. Empty-list check

```
    if (messageSetToWrite.getMessageSetInfo().isEmpty()) {
      throw new IllegalArgumentException("Message write set cannot be empty");
    }
```
The list will never be empty in `put()` because of L1364 in `applyPut`.  The check for empty-list in `put()` is when VCR receives `PUT` from front-end, which is not a goal anymore. As of today, VCR receives `PUT` only from backend-server through replication, which goes through L1364 in `applyPut`.

2. Added check-for-duplicates, although this should be at a higher level in the replication-layer because I see two implementations of `checkDuplicates` at the store level, and now a third one.

Reply:
Ack